### PR TITLE
Update constructors for resources CatalogSrouce, ClusterRoleBinding, Configmap, ControllerRevision, CronJob, CSIStorageCapacity, DataSource, DV and DataImportCron

### PR DESCRIPTION
**Pull request by oharan2**
##### Short description:

This pr covers c-d resources, todo continue with e...z

### CODE REVIEW

File: ocp_resources/catalog_source.py [hunk 1]
```diff
@@ -1,34 +1,35 @@
...
-            name=name,
-            namespace=namespace,
-            client=client,
-            teardown=teardown,
-            yaml_file=yaml_file,
-            delete_timeout=delete_timeout,
-            **kwargs,
-        )
+        """
+        Args:
```
Comment (myakove) [inline ocp_resources/catalog_source.py:22]: Please fix the docstring, seems like the args description don't match the arg

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/catalog_source.py [hunk 2]
```diff
@@ -1,34 +1,41 @@
...
+            source_type (str): Name of the source type, could be one of the following:
+                grpc with an image reference: OLM pulls the image and runs the pod,
+                    which is expected to serve a compliant API.
+                grpc with an address field: OLM attempts to contact
+                    the gRPC API at the given address. This should not be used in most cases.
```
Comment (rnetser) [inline ocp_resources/catalog_source.py:29]: we do not duplicate the api docs; please include the type and short description; in this case `Name of the source type` is enough.

Applies to all places in this PR

File: ocp_resources/catalog_source.py [hunk 3]
```diff
@@ -1,34 +1,41 @@
...
+                configmap: OLM parses config map data and runs a pod that can serve the gRPC API over it.
+            image (str): Index image for the catalog.
```
Comment (rnetser) [inline ocp_resources/catalog_source.py:31]: do not include implementation which is not part of this code

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/catalog_source_config.py [hunk 4]
```diff
@@ -1,35 +1,48 @@
-from ocp_resources.constants import PROTOCOL_ERROR_EXCEPTION_DICT, TIMEOUT_4MINUTES
+from ocp_resources.constants import PROTOCOL_ERROR_EXCEPTION_DICT
```
Comment (rnetser) [inline ocp_resources/catalog_source_config.py:1]: please remove this resource; it was deprecated in 4.5
```
oc get CatalogSourceConfig
error: the server doesn't have a resource type "CatalogSourceConfig"

```

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/catalog_source.py (hunk 3 above)
Comment (oharan2) [reply ocp_resources/catalog_source.py]: I didn't understand what is this referred to.?

File: ocp_resources/catalog_source_config.py (hunk 4 above)
Comment (oharan2) [reply ocp_resources/catalog_source_config.py]: Done

File: ocp_resources/cluster_role_binding.py [hunk 5]
```diff
@@ -13,12 +13,11 @@ class ClusterRoleBinding(Resource):
 
     def __init__(
         self,
```
Comment (rnetser) [inline ocp_resources/cluster_role_binding.py:15]: link to api and docstring are missing

File: ocp_resources/configmap.py [hunk 6]
```diff
@@ -1,34 +1,23 @@
...
-            namespace=namespace,
-            client=client,
-            teardown=teardown,
-            yaml_file=yaml_file,
-            delete_timeout=delete_timeout,
-            **kwargs,
-        )
+        """
+        Args:
+            data (object): contains the configuration data.
```
Comment (rnetser) [inline ocp_resources/configmap.py:18]:
```suggestion
            data (dict): key-value configuration pairs
```

File: ocp_resources/catalog_source.py [hunk 7]
```diff
@@ -1,34 +1,32 @@
...
-            delete_timeout=delete_timeout,
-            **kwargs,
-        )
+        """
+        Args:
+            source_type (str): Name of the source type.
+            image (str): Image index for the catalog.
+            display_name (str): Display name for the catalog in the web console and CLI.
+            publisher (str): Name of the publisher.
+            update_strategy_registry_poll_interval (str): Time interval between checks of the latest
```
Comment (rnetser) [inline ocp_resources/catalog_source.py:26]:
```suggestion
            update_strategy_registry_poll_interval (str, optional): Time interval between checks of the latest
```

please indicate in all places args that are optional (for those that are not optional and we do not have a validation about them in to_dict, please open issues to fix)

File: ocp_resources/data_import_cron.py [hunk 8]
```diff
@@ -25,22 +21,26 @@ def __init__(
...
-            delete_timeout=delete_timeout,
-            **kwargs,
-        )
+        """
+        Args:
+            garbage_collect (str, optional): whether old PVCs should be cleaned up after a new PVC is imported.
+                Options are "Outdated"/"Never". Default value is "Outdated".
+            imports_to_keep (int, optional): number of import PVCs to keep when garbage collecting. Default value is 3.
+            managed_data_source(str, optional): specifies the name of the corresponding DataSource to manage.
+                DataSource has to be in the same namespace. Default value is "".
```
Comment (rnetser) [inline ocp_resources/data_import_cron.py:32]:
```suggestion
                DataSource has to be in the same namespace.
            
```
The code does not set it to `""`, so should not be included here. pleaser remove in all places

File: ocp_resources/data_import_cron.py [hunk 9]
```diff
@@ -25,22 +21,26 @@ def __init__(
...
+            schedule (str, optional): specifies in cron format when and how often to look for new imports.
+                Default value is "".
```
Comment (rnetser) [inline ocp_resources/data_import_cron.py:34]: ditto

File: ocp_resources/data_import_cron.py [hunk 10]
```diff
@@ -25,22 +21,26 @@ def __init__(
...
-            teardown=teardown,
-            privileged_client=privileged_client,
-            yaml_file=yaml_file,
-            delete_timeout=delete_timeout,
-            **kwargs,
-        )
+        """
+        Args:
+            garbage_collect (str, optional): whether old PVCs should be cleaned up after a new PVC is imported.
+                Options are "Outdated"/"Never". Default value is "Outdated".
```
Comment (rnetser) [inline ocp_resources/data_import_cron.py:29]:
```suggestion
                Options are "Outdated"/"Never". 
```
This is not set by the code here

File: ocp_resources/data_source.py [hunk 11]
```diff
@@ -1,33 +1,28 @@
...
-            teardown=teardown,
-            yaml_file=yaml_file,
-            delete_timeout=delete_timeout,
-            **kwargs,
-        )
+        """
+        Args:
+            source (object): Source is the current source of the data referenced by the DataSource object.
+            For more information on DataSourceSource on kubevirt API:
+                https://kubevirt.io/cdi-api-reference/main/definitions.html#_v1beta1_datasourcesource
```
Comment (rnetser) [inline ocp_resources/data_source.py:23]: please remove

File: ocp_resources/data_source.py [hunk 12]
```diff
@@ -1,33 +1,28 @@
...
-            namespace=namespace,
-            client=client,
-            teardown=teardown,
-            yaml_file=yaml_file,
-            delete_timeout=delete_timeout,
-            **kwargs,
-        )
+        """
+        Args:
+            source (object): Source is the current source of the data referenced by the DataSource object.
```
Comment (rnetser) [inline ocp_resources/data_source.py:21]:
```suggestion
            source (dict): The source of the data referenced in the DataSource object.
```
(`object` is too broad)

File: ocp_resources/datavolume.py [hunk 13]
```diff
@@ -99,42 +92,27 @@ def __init__(
...
-            should be bound immediately.
-            preallocation (bool, default: None): preallocate disk space.
-            teardown (bool, default: True): Indicates if this resource would need to be deleted.
-            privileged_client (DynamicClient, default: None): Instance of Dynamic client
-            yaml_file (yaml, default: None): yaml file for the resource.
-            delete_timeout (int, default: 4 minutes): timeout associated with delete action.
-            api_name (str, default: "pvc"): api used for DV, pvc/storage
-            delete_after_completion (str, default: None): annotation for garbage collector - "true"/"false"
+            content_type (str, optional): DataVolume content type. Default value is "kubevirt".
+            access_modes (str, optional): DataVolume access mode. Default value is None.
```
Comment (rnetser) [inline ocp_resources/datavolume.py:100]:
```suggestion
            access_modes (str, optional): DataVolume access mode. 
```
Either keep `(str, default: None)` convention or remove

File: ocp_resources/datavolume.py [hunk 14]
```diff
@@ -176,7 +154,7 @@ def to_dict(self):
                 self.res["spec"]["source"][self.source]["secretRef"] = self.secret.name
             if self.volume_mode:
                 self.res["spec"][self.api_name]["volumeMode"] = self.volume_mode
-            if self.source == "http" or "registry":
+            if self.source == "http" or self.source == "registry":
```
Comment (rnetser) [inline ocp_resources/datavolume.py:157]:
```suggestion
            if self.source in ["http", "registry"]:
```

**Review by rnetser — CHANGES_REQUESTED** [review]

**Comment by rnetser** [comment]
@oharan2 I strongly suggest you split this PR into multiple PRs

**Comment by oharan2** [comment]
Splitting this PR

**Pull request closed without merging**